lib/puppet/indirector/file_metadata/http.rb

```diff
@@ -8,12 +8,12 @@ class Puppet::Indirector::FileMetadata::Http < Puppet::Indirector::GenericHttp
 
   include Puppet::FileServing::TerminusHelper
 
-  @http_method = :head
-
   def find(request)
-    head = super
+    uri = URI(request.uri)
```

Comment:
Really this should be `URI(Puppet::Util.uri_encode(request.uri))`, but that would break someone that's specified `source => 'https://host/foo%20bar'` in their manifest to work around the current behavior. I'll add a ticket for puppet 7

Reply:
LMK when you've opened this ticket. It doesn't look like that's happened yet

Reply:
I filed it as https://tickets.puppetlabs.com/browse/PUP-10492